src/main/java/seedu/modtrek/model/module/Grade.java

```diff
@@ -3,44 +3,40 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.modtrek.commons.util.AppUtil.checkArgument;
 
-import java.util.Arrays;
-import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 /**
- * The type Grade.
+ * Grade denotes the grading based on NUS system for each module.
  */
 public class Grade {
 
-    /**
-     * The constant MESSAGE_CONSTRAINTS.
-     */
     public static final String MESSAGE_CONSTRAINTS =
             "Grade should be either one of [A+, A, A-, B+, B, B-, C+, C, D+, D, F, S, U]";
 
-    private static final List<String> VALID_GRADES = Arrays.asList(new String[] {
-        "A+",
-        "A",
-        "A-",
-        "B+",
-        "B",
-        "B-",
-        "C+",
-        "C",
-        "D+",
-        "D",
-        "F",
-        "S",
-        "U",
-        "" // empty grade field
-    });
+    private static Map<String, Double> gradeToPoints = Map.ofEntries(
+        Map.entry("A+", 5.0),
+        Map.entry("A", 5.0),
+        Map.entry("A-", 4.5),
+        Map.entry("B+", 4.0),
+        Map.entry("B", 3.5),
+        Map.entry("B-", 3.0),
+        Map.entry("C+", 2.5),
+        Map.entry("C", 2.0),
+        Map.entry("D+", 1.5),
+        Map.entry("D", 1.0),
+        Map.entry("F", 0.0),
+        Map.entry("S", 5.0),
+        Map.entry("U", 0.0),
```

Comment:
We can include grades CS and CU too.

Reply:
S and U are the CS/CU grades.

Reply:
Actually there are both S/U and CS/CU if I'm not wrong. S/U is attained by exercising S/U option (limited to 5 in Y1 and 2 in Y2-4) and CS/CU is available for some modules only. In terms of behavior, both are similar but I thought we should provide more grade options for user where possible.

Reply:
Okay sure, though I feel either way is still satisfactory and unsatisfactory. I added it in but I don't think it is that useful. Maybe someone will want to make the distinction between S/U-ed mod and auto S/U mods I guess.

Reply:
Alright. For clarification, we'll specify all possible grades available in our app in our UG.